Kokkos build failure with intel/19 and shared library support: unresolvable R_X86_64_TPOFF32 relocation against symbol

Kokkos builds with intel/19 and shared library support fail with the following type of output:

```
ld: CMakeFiles/KokkosCore_PerformanceTest_Mempool.dir/test_mempool.cpp.o(.text._ZN6Kokkos4Impl21ParallelReduceAdaptorINS_11RangePolicyIJNS_6SerialEN11TestFunctor15TagAllocDeallocEEEES4_lE7executeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKS6_RKS4_Rl[_ZN6Kokkos4Impl21ParallelReduceAdaptorINS_11RangePolicyIJNS_6SerialEN11TestFunctor15TagAllocDeallocEEEES4_lE7executeERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKS6_RKS4_Rl]+0x3e8): unresolvable R_X86_64_TPOFF32 relocation against symbol `_ZN6Kokkos4Impl22SharedAllocationRecordIvvE18t_tracking_enabledE'
ld: final link failed: Nonrepresentable section on output
make[2]: *** [core/perf_test/KokkosCore_PerformanceTest_Mempool] Error 1
make[2]: Leaving directory `/home/ndellin/kokkos/testing/Intel19'
make[1]: *** [core/perf_test/CMakeFiles/KokkosCore_PerformanceTest_Mempool.dir/all] Error 2
```

**Reproducer** (kokkos-dev* modules referenced):
```
module load sems-archive-cmake/3.17.1 sems-archive-gcc/8.3.0 sems-archive-intel/19.0.5

cmake -DKokkos_ENABLE_SERIAL=ON -DCMAKE_CXX_COMPILER=icpc -DBUILD_SHARED_LIBS=ON -DKokkos_ENABLE_TESTS=ON $KOKKOS_PATH
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -202,10 +202,6 @@ KOKKOS_SETUP_BUILD_ENVIRONMENT()
 # 7) Export and install targets
 
 OPTION(BUILD_SHARED_LIBS "Build shared libraries" OFF)
-# Workaround for building position independent code.
-IF(BUILD_SHARED_LIBS)
-  SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
-ENDIF()
 
 SET(KOKKOS_EXT_LIBRARIES Kokkos::kokkos Kokkos::kokkoscore Kokkos::kokkoscontainers Kokkos::kokkosalgorithms)
 SET(KOKKOS_INT_LIBRARIES kokkos kokkoscore kokkoscontainers kokkosalgorithms)

